common/archiver/provider/provider.go

```diff
@@ -64,51 +63,43 @@ type (
 
 // NewArchiverProvider returns a new Archiver provider
 func NewArchiverProvider(
-	historyContainer *archiver.HistoryBootstrapContainer,
-	visibilityContainer *archiver.VisibilityBootstrapContainer,
 	historyArchiverConfigs *HistoryArchiverConfigs,
 	visibilityArchiverConfigs *VisibilityArchiverConfigs,
 ) ArchiverProvider {
 	return &archiverProvider{
-		historyContainer:          historyContainer,
-		visibilityContainer:       visibilityContainer,
 		historyArchiverConfigs:    historyArchiverConfigs,
 		visibilityArchiverConfigs: visibilityArchiverConfigs,
 	}
 }
 
-func (p *archiverProvider) GetHistoryArchiver(scheme, serviceName string, tags ...tag.Tag) (archiver.HistoryArchiver, error) {
-	key := p.getArchiverKey(scheme, serviceName)
-	if historyArchiver, ok := p.historyArchivers[key]; ok {
+func (p *archiverProvider) GetHistoryArchiver(scheme, serviceName string, container *archiver.HistoryBootstrapContainer) (archiver.HistoryArchiver, error) {
```

Comment:
I kind of think it makes sense now to have CreateHistoryArchiver and CreateVisibilityArchiver methods. The reason is its kind of weird to understand when you can or cannot provide BootstrapContainer. For example right now in if archiver already exists and bootstrapContainer is provided the client will get back an archiver but it will not be based on the bootstrap container they provided. Thoughts?